### PR TITLE
fix(popover-manager, tooltip-manager): Properly query referenceElement in shadowRoot

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #3625

## Summary

fix(popover-manager, tooltip-manager): Properly query referenceElement in shadowRoot.

Uses composedPath to get referenceElement and if inside a popover was clicked.

### CODE REVIEW

File: src/components/calcite-popover-manager/calcite-popover-manager.tsx [hunk 1]
```diff
@@ -52,9 +52,22 @@ export class CalcitePopoverManager {
   //
   //--------------------------------------------------------------------------
 
-  getRelatedPopover = (element: HTMLElement): HTMLCalcitePopoverElement => {
-    const { selector, el } = this;
-    const id = element.closest(selector)?.getAttribute(POPOVER_REFERENCE);
```
Comment (driskull) [inline src/components/calcite-popover-manager/calcite-popover-manager.tsx:57]: closest wasn't working in shadowDom cases. Replacing with composedPath fixes this.

**Comment by driskull** [comment]
@jcfranco can you review?

This PR is a dependent of https://github.com/Esri/calcite-components/pull/4034

File: src/components/tooltip-manager/tooltip-manager.tsx [hunk 2]
```diff
@@ -43,9 +43,22 @@ export class TooltipManager {
...
-    const id = element.closest(selector)?.getAttribute(TOOLTIP_REFERENCE);
+  queryTooltip = (composedPath: EventTarget[]): HTMLCalciteTooltipElement => {
+    const { el } = this;
+
+    if (!composedPath.includes(el)) {
+      return null;
+    }
+
+    const referenceElement = (composedPath as HTMLElement[]).find(
+      (pathEl) => pathEl?.hasAttribute && pathEl.hasAttribute(TOOLTIP_REFERENCE)
```
Comment (jcfranco) [inline src/components/tooltip-manager/tooltip-manager.tsx:54]: Suggestion: `pathEl?.hasAttribute?.(TOOLTIP_REFERENCE)`?

**Review by jcfranco — APPROVED** [review]
Other than minor suggestion, this LGTM! 

Before merging, can you add a test for this in case we refactor later on?

**Pull request merged**